### PR TITLE
Move spotify secret and id to a .env file

### PR DESCRIPTION
**Pull request by rainswerld**

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -31,7 +30,7 @@ const App = () => {
         });
       });
     });
-  }, [genres.selectedGenre, spotify.ClientId, spotify.ClientSecret]);
+  }, [genres.selectedGenre, process.env.REACT_APP_SPOTIFY_ID, process.env.REACT_APP_SPOTIFY_SECRET]);
```
Comment (alookatommorow) [inline client/src/App.js]: we shouldn't need to pass the credentials here since they are not going to change

File: client/package.json [hunk 2]
```diff
@@ -37,5 +38,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
+    "dotenv": "^10.0.0"
```
Comment (alookatommorow) [inline client/package.json:43]: we might need this in production too, i am not really sure.

File: client/src/App.js (hunk 1 above)
Comment (rainswerld) [reply client/src/App.js]: good point :) will remove

File: client/package.json (hunk 2 above)
Comment (rainswerld) [reply client/package.json:43]: I think we would, but should we cross that bridge later? or try and figure it out now?

File: client/src/App.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ const App = () => {
         });
       });
     });
-  }, [genres.selectedGenre, process.env.REACT_APP_SPOTIFY_ID, process.env.REACT_APP_SPOTIFY_SECRET]);
+  }, [genres.selectedGenre]);
```
Comment (rainswerld) [inline client/src/App.js:33]: here it is removed since they won't change.

**Comment by rainswerld** [comment]
Are we good to merge?

**Comment by rainswerld** [comment]
I actually opened to app again today and got the 400 error I was seeing yesterday when trying to connect to Spotify. When I moved the .env file back to the root directory, everything worked again. Not sure how we should work around that...

**Review by alookatommorow — APPROVED** [review]
LGTM, add .env to .gitignore

**Pull request merged**